### PR TITLE
[receiver/dockerstats] adding renamed cpu_utilization metric

### PR DESCRIPTION
**Pull request by carlossscastro**
**Description:**
This change starts the process for deprecating the metric `container.cpu.percent` in favor of `container.cpu.utilization` to adhere to semantic convention.

At this step introducing the new metric name disabled by default, leaving the old metric enabled

**Link to tracking Issue:** #21807

**Testing:** 
```
❯ go clean -testcache
❯ make test
go test -race -timeout 300s -parallel 4 --tags="" ./...
ok  	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/dockerstatsreceiver	0.759s
ok  	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/dockerstatsreceiver/internal/metadata	0.325s
❯ go test --tags integration ./...
ok  	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/dockerstatsreceiver	16.050s
ok  	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/dockerstatsreceiver/internal/metadata	0.209s
```

**Documentation:** 
Metrics documentation was updated using `mdatagen`

### CODE REVIEW

**Comment by carlossscastro** [comment]
@atoulme
New PR with the changes suggested in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/22159

**Comment by songy23** [comment]
cc @gbbr

File: receiver/dockerstatsreceiver/metadata.yaml [hunk 1]
```diff
@@ -121,9 +121,15 @@ metrics:
...
+    enabled: false
+    description: "Percent of CPU used by the container."
+    unit: "1"
+    gauge:
+      value_type: double
   container.cpu.percent:
     enabled: true
-    description: "Percent of CPU used by the container."
+    description: "Deprecated: use `container.cpu.utilization` metric instead. Percent of CPU used by the container."
     unit: "1"
```
Comment (dmitryax) [inline receiver/dockerstatsreceiver/metadata.yaml:133]:
```suggestion
    unit: "1"
    warnings:
      if_enabled: This metric will be disabled in v0.82.0 and removed in v0.85.0.
```

File: receiver/dockerstatsreceiver/metadata.yaml [hunk 2]
```diff
@@ -121,9 +121,15 @@ metrics:
       value_type: int
       monotonic: true
       aggregation: cumulative
+  container.cpu.utilization:
+    enabled: false
+    description: "Percent of CPU used by the container."
+    unit: "1"
```
Comment (dmitryax) [inline receiver/dockerstatsreceiver/metadata.yaml:127]:
```suggestion
    unit: "1"
    warnings:
      if_enabled_not_set: This metric will be enabled by default in v0.82.0.
```

File: receiver/dockerstatsreceiver/metadata.yaml (hunk 1 above)
Comment (carlossscastro) [reply receiver/dockerstatsreceiver/metadata.yaml:135]: Addressed in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/22823/commits/551b12a69c22cd091e76c41270ee681d9b1cdfab

File: receiver/dockerstatsreceiver/metadata.yaml (hunk 2 above)
Comment (carlossscastro) [reply receiver/dockerstatsreceiver/metadata.yaml:127]: Addressed in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/22823/commits/551b12a69c22cd091e76c41270ee681d9b1cdfab

Comment (carlossscastro) [reply receiver/dockerstatsreceiver/metadata.yaml:127]: Thanks!

**Review by dmitryax — APPROVED** [review]

**Comment by dmitryax** [comment]
Thanks, @carlossscastro!

**Pull request merged**